lib/elixir_bench/github/client.ex

```diff
@@ -1,44 +1,9 @@
 defmodule ElixirBench.Github.Client do
-  alias ElixirBench.Github.Client
 
-  defstruct [:base_url, ssl_options: []]
 
-  def get_plain(%Client{} = client, url, params \\ []) do
-    get(client, url(client, url, params), [{"accept", "text/plain"}], &(&1))
-  end
+  @callback get_plain(path :: term) :: {:ok, config :: term} | {:error, reason :: term}
 
-  def get_json(%Client{} = client, url, params \\ []) do
-    get(client, url(client, url, params), [{"accept", "application/json"}], &Antidote.decode/1)
-  end
+  @callback get_json(path :: term) :: {:ok, config :: term} | {:error, reason :: term}
 
-  def get_yaml(%Client{} = client, url, params \\ []) do
-    get(client, url(client, url, params), [{"accept", "application/yaml"}], &decode_yaml/1)
-  end
-
-  @doc false
-  # Public for testing purposes
-  def decode_yaml(content) do
-    case :yamerl.decode(content, [:str_node_as_binary, map_node_format: :map]) do
-      [parsed] -> {:ok, parsed}
-      _ -> {:error, :invalid}
-    end
-  end
-
-  defp get(client, url, headers, cb) do
-    ssl_options = client.ssl_options
-    options = [:with_body, ssl_options: ssl_options]
-
-    case :hackney.get(url, headers, <<>>, options) do
-      {:ok, 200, _headers, data} ->
-        cb.(data)
-      {:ok, status, _headers, data} ->
-        {:error, {status, data}}
-      {:error, error} ->
-        {:error, error}
-    end
-  end
-
-  defp url(%{base_url: base}, url, params) do
-    Path.join(base, url) <> "?" <> URI.encode_query(params)
-  end
+  @callback get_yaml(path :: term) :: {:ok, config :: term} | {:error, reason :: term}
```

Comment:
if I see this correctly currently only `get_yaml` is used so we could even drop the other methods.

Am I missing something? @michalmuskala